src/transformers/trainer.py

```diff
@@ -995,8 +995,8 @@ def _wrap_model(self, model, training=True):
                 find_unused_parameters = True
             model = nn.parallel.DistributedDataParallel(
                 model,
-                device_ids=[self.args.local_rank],
-                output_device=self.args.local_rank,
+                device_ids=[self.args.local_rank] if self.args.n_gpu > 1 else None,
+                output_device=self.args.local_rank if self.args.n_gpu > 1 else None,
```

Comment:
This is not a good check of whether we are on distributed training CPU: `self.args.n_gpu` will be 1 if the training is distributed on several GPUs (a bit counter-intuitive I know, but it was like this at the start of `Trainer` and we didn't change it to avoid breaking changes).

The proper check is `self.args._n_gpu != 0`: this variable is only 0 on CPU or TPU, but TPU is already excluded here.

Reply:
thanks, updated accordingly.